app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java

```diff
@@ -392,25 +380,34 @@ public void loadDocument() {
                 sel[0] = Math.min(sel[0], content.length());
                 sel[1] = Math.min(sel[1], content.length());
 
+                if (_editTextUndoRedoHelper != null) {
+                    _editTextUndoRedoHelper.disconnect();
+                    _editTextUndoRedoHelper.clearHistory();
+                }
+
                 _hlEditor.withAutoFormatDisabled(() -> _hlEditor.setText(content));
 
+                if (_editTextUndoRedoHelper == null) {
```

Comment:
create the UndoRedoHelper _after_ first load. Also disable and reset it after a load.